lib/vega_lite/convert.ex

```diff
@@ -1,2 +1,223 @@
 defmodule VegaLite.Convert do
+  @moduledoc """
+  Various export methods for a `VegaLite` specification.
+
+  All of the exports are performed via a Rustler NIF that wraps the
+  [vl-convert-rs](https://github.com/vega/vl-convert) Rust library.
+  """
+
+  alias VegaLite.Convert.WxViewer
+  alias VegaLite.Native
+
+  @doc """
+  Saves a `VegaLite` specification to file in one of
+  the supported formats.
+
+  Any additional options provided beyond `:format` are passed to
+  the functions that export to the desired format.
+
+  ## Options
+
+    * `:format` - the format to export the graphic as,
+      must be either of: `:json`, `:html`, `:png`, `:svg`, `:pdf`,
+      `:jpeg`, `:jpg`. By default the format is inferred from the
+      file extension.
+  """
+  @spec save!(VegaLite.t(), binary(), keyword()) :: :ok
+  def save!(vl, path, opts \\ []) do
+    {format, opts} =
+      Keyword.pop_lazy(opts, :format, fn ->
+        path
+        |> Path.extname()
+        |> String.trim_leading(".")
+        |> String.to_existing_atom()
+      end)
+
+    content =
+      case format do
+        :json ->
+          to_json(vl, opts)
+
+        :html ->
+          to_html(vl, opts)
+
+        :png ->
+          to_png(vl, opts)
+
+        :svg ->
+          to_svg(vl)
+
+        :pdf ->
+          to_pdf(vl)
+
+        jpeg when jpeg in [:jpeg, :jpg] ->
+          to_jpeg(vl, opts)
+
+        _ ->
+          raise ArgumentError,
+                "unsupported export format, expected :json, :html, :png, :svg, :pdf, :jpeg or :jpg got: #{inspect(format)}"
+      end
+
+    File.write!(path, content)
+  end
+
+  @doc """
+  Returns the underlying Vega-Lite specification as JSON.
+
+  ## Options
+
+    * `:target` - specifies whether JSON export is in the VegaLite
+    format or Vega. Valid options are `:vega_lite` or `:vega`. Default
+    is `:vega_lite`.
+  """
+  @spec to_json(vl :: VegaLite.t(), opts :: keyword()) :: String.t()
+  def to_json(vl, opts \\ []) do
+    vega_lite_json =
+      vl
+      |> VegaLite.to_spec()
+      |> Jason.encode!()
+
+    case Keyword.get(opts, :target, :vega_lite) do
+      :vega_lite ->
+        vega_lite_json
+
+      :vega ->
+        {:ok, vega_json} = Native.vegalite_to_vega(vega_lite_json)
+        vega_json
```

Comment:
If the NIF returns an error, we will get a match error, so instead let's do this:

```suggestion
        Native.vegalite_to_vega(vega_lite_json) |> unwrap!()
```

where

```elixir
defp unwrap!(:ok), do: :ok
defp unwrap!({:ok, value}), do: value
defp unwrap!({:error, error}), do: raise(List.to_string(error))
```

and we can use it for all NIF calls :)